lib/routes/apnews/api.ts

```diff
@@ -0,0 +1,57 @@
+import { Route, ViewType } from '@/types';
+import { fetchArticle } from './utils';
+import ofetch from '@/utils/ofetch';
+import timezone from '@/utils/timezone';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/api/:tags?',
+    categories: ['traditional-media', 'popular'],
+    example: '/apnews/api/business',
+    view: ViewType.Articles,
+    parameters: {
+        tags: {
+            description: 'See https://github.com/kovidgoyal/calibre/blob/81666219718b5f57d56b149a7ac017cc2a76b931/recipes/ap.recipe#L43-L46',
+            default: 'apf-topnews',
+        },
+    },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['apnews.com/'],
+        },
+    ],
+    name: 'News',
+    maintainers: ['zoenglinghou', 'mjysci', 'TonyRL'],
+    handler,
+};
+
+async function handler(ctx) {
+    const { tags = 'apf-topnews' } = ctx.req.param();
+    const apiRootUrl = 'https://afs-prod.appspot.com/api/v2/feed/tag';
+    const url = `${apiRootUrl}?tags=${tags}`;
+    const res = await ofetch(url);
+
+    const list = res.cards
+        .map((e) => ({
+            title: e.contents[0].headline,
+            link: e.contents[0].localLinkUrl,
```

Comment:
Not every `cards` has `contents[0]`

<details>
<summary>Details</summary>

![image](https://github.com/user-attachments/assets/bb5611ea-0de1-48e3-930e-10573c71e073)


</details>

Reply:
What exactly is the content of the whole response?